base/src/com/google/idea/blaze/base/bazel/LocalInvoker.java

```diff
@@ -0,0 +1,275 @@
+import com.google.idea.blaze.base.scope.BlazeContext;
+import com.google.idea.blaze.base.scope.output.IssueOutput;
+import com.google.idea.blaze.base.settings.Blaze;
+import com.google.idea.blaze.base.settings.BlazeUserSettings;
+import com.google.idea.blaze.base.settings.BuildBinaryType;
+import com.google.idea.blaze.common.PrintOutput;
+import com.google.idea.blaze.exception.BuildException;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+
+/**
+ * A local Blaze/Bazel invoker that issues commands via CLI.
+ */
+public class LocalInvoker extends AbstractBuildInvoker1 {
+  private static Logger logger = Logger.getInstance(LocalInvoker.class);
+  private final BuildBinaryType buildBinaryType;
+
+  public LocalInvoker(Project project, BlazeContext blazeContext, BuildSystem buildSystem, BuildBinaryType binaryType) {
+    super(project, blazeContext, buildSystem, binaryPath(binaryType, project));
+    this.buildBinaryType = binaryType;
+  }
+
+  @Override
+  public BuildResultHelper createBuildResultHelper() {
+    return new BuildResultHelperBep();
+  }
+
+  @Override
+  public BlazeCommandRunner getCommandRunner() {
+    return new CommandLineBlazeCommandRunner();
+  }
+
+  @Override
+  public boolean supportsHomeBlazerc() {
+    return true;
+  }
+
+  @Override
+  public boolean supportsParallelism() {
+    return false;
+  }
+
+  @Override
+  public ImmutableSet<Capability> getCapabilities() {
+    return ImmutableSet.of(Capability.IS_LOCAL, Capability.SUPPORTS_CLI);
+  }
+
+  @Override
+  public BuildEventStreamProvider invoke(BlazeCommand.Builder blazeCommandBuilder)
+    throws BuildException {
+    try {
+      performGuardCheck(project, blazeContext);
+    }
+    catch (ExecutionDeniedException e) {
+      throw new BuildException(e.getMessage(), e);
+    }
+    File outputFile = BuildEventProtocolUtils.createTempOutputFile();
+    BuildResult buildResult =
+      issueBuild(blazeCommandBuilder, WorkspaceRoot.fromProject(project), blazeContext, outputFile);
+    if (blazeCommandBuilder.build().getName().equals(BlazeCommandName.BUILD)) {
+      BuildDepsStatsScope.fromContext(blazeContext)
+        .ifPresent(stats -> stats.setBazelExitCode(buildResult.exitCode));
+    }
+    return getBepStream(outputFile);
+  }
+
+  @Override
+  public InputStream invokeQuery(BlazeCommand.Builder blazeCommandBuilder) {
+    try {
+      performGuardCheck(project, blazeContext);
+    }
+    catch (ExecutionDeniedException e) {
+      logger.error(e);
```

Comment:
This looks like error swallowing to me.

Reply:
Yes, not perfect but `logger.error` should be fine. Also, not worth deviating from AOSP for this in my opinion.